.github/workflows/argus-helm-chart-update.yaml

```diff
@@ -0,0 +1,111 @@
+  update-helm-deps:
+    name: Update Helm Dependencies
+    needs: [fetch-app-info]
+    runs-on: ubuntu-latest
+    if: needs.fetch-app-info.outputs.app_info.envs != '[]'
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        env: ${{ fromJson(needs.fetch-app-info.outputs.app_info).envs }}
+    steps:
+      - name: Log info
+        run: |
+          echo "Updating chart in env: ${{ matrix.env.name }} - app root dir: ${{ fromJson(needs.fetch-app-info.outputs.app_info).project_root_path }}"
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup updatecli
+        uses: updatecli/updatecli-action@v2
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.CZI_RELEASE_PLEASE_APP_ID }}
+          private_key: ${{ secrets.CZI_RELEASE_PLEASE_PK }}
+
+      - name: Write updatecli manifest
+        id: write_manifest
+        run: |
+          export UPDATECLI_MANIFEST_FILE=updatecli-${{ matrix.env.name }}.yaml
+          echo "UPDATECLI_MANIFEST_FILE=$UPDATECLI_MANIFEST_FILE" >> $GITHUB_OUTPUT
+          cat <<EOF > $UPDATECLI_MANIFEST_FILE
+          name: Helm chart update
+
+          # Defines how to get "source" information such as Jenkins version
+          sources:
+            stackHelmChart:
+              name: Get the latest Helm chart version
+              kind: helmchart
+              spec:
+                url: https://chanzuckerberg.github.io/argo-helm-charts
+                name: stack
+
+          # Defines "targets" which need to be updated if different than "source" information.
+          targets:
+            helmChartVersions:
+              name: Bump Helm chart versions
+              kind: yaml
+              spec:
+                engine: yamlpath
+                key: $.dependencies[?(@.name == 'stack')].version
+                file: ${{ fromJson(needs.fetch-app-info.outputs.app_info).project_root_path }}/.infra/${{ matrix.env.name }}/Chart.yaml
+          EOF
+
+      - name: Run updatecli
+        id: run_updatecli
+        run: |
+          export UPDATE_LOG_FILE=$(uuidgen)-update.log
+          echo "Running updatecli with manifest ${{ steps.write_manifest.outputs.UPDATECLI_MANIFEST_FILE }}"
+          cat ${{ steps.write_manifest.outputs.UPDATECLI_MANIFEST_FILE }}
+
+          echo "----------------------------------------"
+          updatecli apply --config ${{ steps.write_manifest.outputs.UPDATECLI_MANIFEST_FILE }} >> $UPDATE_LOG_FILE
+          cat $UPDATE_LOG_FILE
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "update_logs<<$EOF" >> $GITHUB_OUTPUT
+          echo "$(cat $UPDATE_LOG_FILE)" >> $GITHUB_OUTPUT
+          echo "$EOF" >> $GITHUB_OUTPUT
+          rm $UPDATE_LOG_FILE
+          rm ${{ steps.write_manifest.outputs.UPDATECLI_MANIFEST_FILE }}
+
+      - name: Create Pull Request
```

Comment:
in your example run the workflow is triggered via cron. is the system somehow able to recognize it shouldn't create a dupe PR every cron trigger?

Reply:
Yeah, `peter-evans/create-pull-request` won't create a PR or commit if there aren't any changes and if there's an existing PR proposing an upgrade to 2.6.0, for example, and the next run finds that 2.7.0 is available it will update the existing PR